dali/operators/input/video_input.cc

```diff
@@ -0,0 +1,165 @@
+    auto &thread_pool = ws.GetThreadPool();
+    this->ForwardCurrentData(encoded_videos, thread_pool);
+
+    // Creating FramesDecoders
+    auto sample = encoded_videos[0];
+    auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
+    size_t size = sample.shape().num_elements();
+    frames_decoders_[0] = std::make_unique<FramesDecoder>(data, size, false);
+
+    assert(output_descs_.empty());
+    DetermineOutputDescs(static_cast<int>(frames_decoders_[0]->NumFrames()));
+
+    // This has to be done for every video file, since we need to know the shape of the frames.
+    if (last_sequence_policy_ == "pad") {
+      InitializePadValue(0);
+    }
+
+    valid_ = true;
+  }
+  output_desc.resize(1);
+  output_desc[0] = output_descs_.front();
+  output_descs_.pop_front();
+  return true;
+}
+
+
+template<>
+void VideoInput<CPUBackend>::RunImpl(Workspace &ws) {
+  auto &output = ws.Output<CPUBackend>(0);
+  output.SetLayout("FHWC");
+  bool full_sequence;
+  for (int64_t s = 0; s < output.num_samples(); s++) {
+    auto pad_value =
+            last_sequence_policy_ == "pad" ? std::optional<SampleView<CPUBackend>>(GetPadFrame())
+                                           : std::nullopt;
+    full_sequence = DecodeFrames(output[s], 0, frames_per_sequence_, pad_value);
+    if (!full_sequence) {
+      break;
+    }
+  }
+  if (!full_sequence || frames_decoders_[0]->NextFrameIdx() == -1) {
+    Invalidate();
+  }
+}
+
+
+DALI_SCHEMA(experimental__inputs__Video)
+                .DocStr(
+                        R"code(
+Streams and decodes a video from a memory buffer. To be used with long and high resolution videos.
+
+Returns a batch of sequences of frames, with the layout: ``(F, H, W, C)``, where:
+
+* ``F`` - number of frames in a sequence,
+* ``H`` - height of the frame,
+* ``W`` - width of the frame,
+* ``C`` - number of channels in the frame.
+
+When using ``fn.inputs.video`` operator inside the DALI Pipeline, the user needs to provide the data
+using :meth:`Pipeline.feed_input`. When the Operator is fed with data, the Pipeline can be run
+multiple times and the ``fn.inputs.video`` operator will return consecutive sequences, as long as
+there is enough data to decode. When the source of the frames (the video file) depletes, user needs
+to call another ``feed_input`` again to provide the next video file to the operator. This Operator
+has an inner-queue for the data, so the ``feed_input`` may be called multiple times and when given
+video file ends, the Operator will fetch the next one automatically from the top of the queue.
+Running the pipeline while there is no data for the ``fn.inputs.video`` to run results in an error.
+
+This operator takes only one video as and input (i.e. ``input_batch_size=1``) and will return
+batches of sequences. Every output batch will have the ``max_batch_size``, set during the Pipeline
+creation. When the number of frames in the video file does not allow to split the frames uniformly
```

Comment:
```suggestion
batches of sequences. Every output batch will consist of ``max_batch_size`` samples (the ``max_batch_size` is set during the Pipeline creation). When the number of frames in the video file does not allow to split the frames uniformly
```

The exact wording in not that important, but if I understand what you mean here, the word `samples` is missing, right?

Reply:
I intended `max_batch_size size`, but samples works also good 👍🏻